strawberryfields/ops.py

```diff
@@ -1910,6 +1910,23 @@ def _decompose(self, reg, **kwargs):
         ]
 
 
+class sMZgate(Gate):
+    r"""Symmetric Mach-Zehnder interferometer"""
+    ns = 2
+
+    def __init__(self, phi_in, phi_ex):
+        super().__init__([phi_in, phi_ex])
+
+    def _decompose(self, reg, **kwargs):
+        # into local phase shifts and two 50-50 beamsplitters
+        return [
+            Command(BSgate(np.pi / 4, np.pi / 2), reg),
+            Command(Rgate(self.p[1] - np.pi / 2), reg[1]),
+            Command(Rgate(self.p[0] - np.pi / 2), reg[0]),
+            Command(BSgate(np.pi / 4, np.pi / 2), reg),
+        ]
```

Comment:
you will need to add an extra test for this

Reply:
(to summerise our slack conversation) I think we will leave this as it is. The current way we do this is more relevant if you want to program a chip. If a user cares more about running simulations quickly, and doesn't care about the physical layout, they can use ``mesh='rectangular'`` instead.